util/paused/paused.go

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This is a copy of the paused helpers from cluster-api, without the
+// v1beta2conditions requirement. Once the v1beta2conditions migration is
+// complete, we will want to use the upstream package.
```

Comment:
It would be worth creating an issue for this so that we don't forget it.

Reply:
I've created #5399  :)